clap_mangen/src/lib.rs

```diff
@@ -221,7 +221,14 @@ impl Man {
 
     fn _render_synopsis_section(&self, roff: &mut Roff) {
         roff.control("SH", ["SYNOPSIS"]);
-        render::synopsis(roff, &self.cmd);
+        if self.cmd.is_flatten_help_set() {
+            for sc in self.cmd.get_subcommands() {
+                render::synopsis(roff, sc);
+                roff.control("br", []);
+            }
```

Comment:
There are times when you should sow the usage for the current command

See https://github.com/clap-rs/clap/blob/d2874a50cf594ace49a9dfa644e734f83491a403/clap_builder/src/output/usage.rs#L107-L113

In general, I'd recommend checking how we use flatten_help to change output
- https://github.com/clap-rs/clap/blob/master/clap_builder/src/output/help_template.rs
- https://github.com/clap-rs/clap/blob/master/clap_builder/src/output/usage.rs

Reply:
I tried to adopt the style

Reply:
I am  not seeing a call to `is_subcommand_required_set` or `is_args_conflicts_with_subcommands_set` to handle this

Reply:
Added